pages/index.js

```diff
@@ -1,19 +1,78 @@
 import ProductsList from "@/components/ProductsList";
 import Icon from "@/components/Icons";
 import { StyledCreateLink } from "@/components/Buttons";
+import styled from "styled-components";
+import { useEffect, useState } from "react";
+
+const StyledSearchForm = styled.form`
+  display: flex;
+  margin-block-start: 30px;
+  border-radius: 5px;
+  width: 100%;
+  border: 1px solid;
+  box-shadow: inset 0px 0px 2px var(--shadowColor);
+  padding-inline: 10px;
+  background-color: white;
+  &:focus-within {
+    border: 2px solid var(--accentColor);
+  }
+`;
+
+const StyledSearchInput = styled.input`
+  background: transparent;
+  border: none;
+  outline: none;
+  padding: 10px;
+  width: 100%;
+  font-size: 16px;
+`;
 
 export default function HomePage({
   products,
   onDeleteProduct,
   onToggleShoppingList,
 }) {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [foundProducts, setFoundProducts] = useState();
+  useEffect(() => {
+    setFoundProducts(
+      products.filter((product) =>
+        product.name.toLowerCase().includes(searchTerm.toLowerCase())
+      )
+    );
+  }, [searchTerm, products]);
+
   return (
     <main>
+      <StyledSearchForm>
+        <Icon variant="search" />
+        <StyledSearchInput
+          onChange={(event) => setSearchTerm(event.target.value)}
+          type="text"
+          id="searchTerm"
+          name="searchTerm"
+          placeholder="Search a product"
+        ></StyledSearchInput>
+      </StyledSearchForm>
+      <h2>All Products</h2>
       <ProductsList
-        products={products}
+        products={searchTerm.length === 0 ? products : foundProducts}
         onDeleteProduct={onDeleteProduct}
         onToggleShoppingList={onToggleShoppingList}
       ></ProductsList>
+      {!products && (
+        <>
+          <p>Currently, your product list is empty. </p>
+          <p>You can add products to your list via the the green + button</p>
```

Comment:
A small typo (two times "the"). 

I didn't get the meaning of the "+" at first sight. Maybe you might put it in quotation marks or replace it with a little plus icon as representation. 
Or even better; You could instead also put a call-to-action button here "Create your first product" that links to the "Create" page. This way you don't have to instruct the user and ask him/her to get started right away.